### ISSUE TITLE
1.1.x fails to restore the cache

### ISSUE THREAD

**Issue by haizaar** [nix-actions/cache-install#17]
I've been a happy use of `cache-install` action on 1.0.x version. Trying 1.1.1 today I see that it can create the cache just fine but it fails to restore it:

```
...
Running set_nix_path from core.sh
Running set_paths from core.sh
Running set_nix_profile_symlink from core.sh
rm: cannot remove '/home/runner/.nix-profile/bin/gsutil': Permission denied
rm: cannot remove '/home/runner/.nix-profile/bin/envsubst': Permission denied
rm: cannot remove '/home/runner/.nix-profile/bin/git-credential-gcloud.sh': Permission denied
rm: cannot remove '/home/runner/.nix-profile/bin/docker-credential-gcloud': Permission denied
rm: cannot remove '/home/runner/.nix-profile/bin/terraform': Permission denied
rm: cannot remove '/home/runner/.nix-profile/bin/bq': Permission denied
rm: cannot remove '/home/runner/.nix-profile/bin/gcloud': Permission denied
```

Switching back to 1.0.8 makes everything to work again.

Can you please assist?

**Comment by rikhuijzer** [comment]
Thank you for opening an issue. This project is pretty difficult to debug so it's always nice to see when someone is using it and finding bugs.

It's a bit of a weird issue because it doesn't happen in the test sets and I don't know why. If you want, you can take a look at https://github.com/rikhuijzer/cache-install/blob/master/.github/workflows/test.yml and see whether you can tell what should be changed to the tests to catch your bug.

Regardless, I've made an attempt to fix the bug in a new branch called `permissions`. You can see the changes in https://github.com/rikhuijzer/cache-install/pull/18. Can you try to run that via:
```yaml
- uses: rikhuijzer/cache-install@permissions
  with:
    [...]
```
and tell me whether it worked?

**Comment by haizaar** [comment]
Thanks for a quick response. Somehow it ended up in a bin - apologies for the lag.

I've tried the fix but it didn't help - same issue:
![image](https://github.com/rikhuijzer/cache-install/assets/58201/2d9937a0-884e-4f5c-96fe-96eefd0cdef5)


And this is after triggering cache rebuild and then re-running the action:
![image](https://github.com/rikhuijzer/cache-install/assets/58201/add7ea39-eba4-4f11-a37a-8bb79a4e0d13)


For the reference, I embed it into custom action:
```yaml
name: Setup Environment
description: Setup Nix env and init Terraform
runs:
  using: composite

  steps:

  - name: Federation auth
    uses: google-github-actions/auth@v1.1.1
    with:
      token_format: access_token  # Must be defined for ${{ steps.auth.outputs.access_token }} to be populated
      workload_identity_provider: projects/....
      service_account: terraform@...

  # Refernce: https://github.com/rikhuijzer/cache-install/issues/1#issuecomment-1316507513
  - name: Install Nix + packages dependencies; cache results
    uses: rikhuijzer/cache-install@permissions  # 1.1x is broken: https://github.com/rikhuijzer/cache-install/issues/17
    with:
      key: nix-${{ hashFiles('.nix/**') }}
      nix_file: ".nix/action-deps.nix"

  # Now with nix installed we can run make as if we are inside nix-shell

  - name: Init TF
    run: make init
    shell: bash
```

Which I then simply call from a workflow as:
```yaml
...
jobs:
  foo:

    name: foo
    permissions:
      id-token: write
      contents: read
    runs-on: ubuntu-latest

    steps:

    - name: Checkout
      uses: actions/checkout@v3

    - name: Setup environment
      uses: ./.github/actions/setup-env

    ...

**Comment by haizaar** [comment]
The `ownership of '/home/runner/.nix-profile' retained as runner:docker` looks highly suspicious. Mb running `rm` with sudo?

**Comment by haizaar** [comment]
[This](https://github.com/rikhuijzer/cache-install/commit/43030ddf0b644d4e63ce2f1fc4f5838eb3a9610a) works better.

However now when I finally got it to work I see a warning about it failing to save the cache:

TBH, I'm not sure why it tried to save the cache in the first place - there was a cache-_hit_ from the get go. From what I can tell, it didn't try to save the cache after hits on 1.0.x.

![image](https://github.com/rikhuijzer/cache-install/assets/58201/9e2de5f9-e907-4fa1-917c-15fc1eae36c8)


Any ideas?

**Comment by rikhuijzer** [comment]
Both problems should now be fixed in v1.1.2. If anything else pops up then do open an issue. They are much appreciated!

**Comment by haizaar** [comment]
v1.1.2 works great indeed! Thank you again for the quick fix!
